src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,257 @@
+        const form = e.currentTarget.parentNode;
+        let token = localStorage.getItem('token');
+        let user = localStorage.getItem('user');
+        // Make use of Bootstraps built in validation, changing the state validated variable to true to indicate that the form has undergone validation (not to indicate if it's passed validation or not)
+        if (form.checkValidity() === false) {
+            e.preventDefault();
+            e.stopPropagation();
+            this.setState({ validated: true });
+        } else {
+            e.preventDefault();
+            this.setState({ validated: true });
+            // If validation passed, then make a put request to the API, updating all the details on the form (which are now stored in the state variables thanks to the handleFieldChange function)
+            axios.put(`https://t-dogg-movies-api.herokuapp.com/users/${user}`, {
+                Username: this.state.username,
+                Password: this.state.password,
+                Email: this.state.email,
+                Birthday: this.state.birthDate
+            }, {
+                headers: { Authorization: `Bearer ${token}`}
+            }).then(response => {
+                const data = response.data;
+                // Update localStorage with the new username
+                localStorage.setItem('user', data.Username);
+                // Reload the page to make sure that the user can immediately start using their new details
+                window.open(`/users/${data.Username}`, '_self');
+            }).catch(error => {
+                console.log('error updating user details')
+            });
+        }
+    };
+
+    // Function for updating the appropriate state variables as the user fills out the update form
+    handleFieldChange(event) {
+        let {name, value} = event.target;
+        this.setState({ [name]: value})
+    }
+
+    // Function to show the modal that confirms you want to delete a user profile
+    showModal() {
+        this.setState({ modalState: true });
+    }
+
+    // Fuunction for closing the modal that confirms you want to delete a user profile
+    closeModal() {
+        this.setState({ modalState: false });
+    }
+
+    // Function for deleting user details. A delete request is made ot the API for this user
+    deleteUserDetails() {
+        let token = localStorage.getItem('token');
+        let user = localStorage.getItem('user');
+        axios.delete(`https://t-dogg-movies-api.herokuapp.com/users/${user}`, {
+            headers: { Authorization: `Bearer ${token}`}
+        }).then(response => {
+            const data = response.data;
+            alert(user + " has been deleted");
+            // Remove the user details and auth token from localStorage, and send the user back to the login page (since they are now logged out)
+            localStorage.removeItem('user');
+            localStorage.removeItem('token');
+            window.open('/', '_self');
```

Comment:
Don't reload the page in a SPA. Use React Router or conditional rendering based on state instead